core/crypto/src/key_file.rs

```diff
@@ -28,10 +28,10 @@ impl KeyFile {
         file.write_all(str.as_bytes())
     }
 
-    pub fn from_file(path: &Path) -> Self {
-        let mut file = File::open(path).expect("Could not open key file.");
+    pub fn from_file(path: &Path) -> std::io::Result<Self> {
```

Comment:
```suggestion
    pub fn from_file(path: &Path) -> io::Result<Self> {
```

That's the idiomatic way to spell io result